ui/src/main/java/com/hyperwallet/android/ui/view/widget/DateWidget.java

```diff
@@ -111,4 +102,37 @@ public String getValue() {
     public void showValidationError(String errorMessage) {
         mTextInputLayout.setError(errorMessage);
     }
+
+    private void showDateSelectDialog() {
+        Calendar calendar = mDateUtil.convertDateFromServerFormatToCalendar(mValue);
+        DatePickerDialog datePickerDialog = new DatePickerDialog(mContext, this,
+                calendar.get(Calendar.YEAR), calendar.get(Calendar.MONTH), calendar.get(Calendar.DAY_OF_MONTH));
+
+        datePickerDialog.setButton(DialogInterface.BUTTON_NEGATIVE, mContext.getString(R.string.cancel_button_label),
+                new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                        if (which == DialogInterface.BUTTON_NEGATIVE) {
+                            if (isValid()) {
+                                mTextInputLayout.setError(null);
+                            }
+                        }
+                    }
+                });
+        datePickerDialog.show();
+    }
+
+    @Override
+    public void onDateSet(DatePicker view, int year, int month, int dayOfMonth) {
+        if (isValid()) {
+            mTextInputLayout.setError(null);
+        }
+
+        mValue = mDateUtil.buildDateFromDateDialogToServerFormat(year, month, dayOfMonth);
+        mEditText.setText(mDateUtil.buildDateFromDateDialogToWidgetFormat(year, month, dayOfMonth));
+        if (isValid()) {
```

Comment:
why do we need to invoke `isValid` twice one in line 127 and here?

Reply:
@peter-joseph Thank you, removed